File: 20170220/NavigationController/NavigationController/ViewController.m
//
//  ViewController.m
//  NavigationController
//
//  Created by ji jun young on 2017. 2. 20..
//  Copyright © 2017년 Fastcampus. All rights reserved.
//

#import "ViewController.h"
#import "ViewControllerSecond.h"

@interface ViewController ()
<UITextFieldDelegate>
//텍스트뷰 델리게이트 선언은 기존 UIScrollview 델리게이트 선언 뒤에 , 하고 하면 됨

@property NSInteger sum;
@property UILabel *sumLabel;
@property UITextField* id;
@property UITextField* pw;
@property UIScrollView* mainScrollView;

@end

@implementation ViewController

//자판기 뷰 앞의 로그인 뷰를 복사해서 약간 수정함
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    CGFloat viewSizeWidth;
    viewSizeWidth = self. view.frame.size.width;
    CGFloat viewSizeHeight;
    viewSizeHeight = self. view.frame.size.height;
    
    self.mainScrollView = [[UIScrollView alloc] initWithFrame:CGRectMake(0, 0, self.view.frame.size.width, self.view.frame.size.height)];
    [self.mainScrollView setContentSize:CGSizeMake(self.mainScrollView.frame.size.width * 2, self.mainScrollView.frame.size.height)];
    [self.mainScrollView setBackgroundColor:[UIColor orangeColor]];
    self.mainScrollView.pagingEnabled = YES;
    //    mainScrollView.delegate = self;
    self.mainScrollView.scrollEnabled = NO;
    [self.view addSubview:self.mainScrollView];
    
    UIView *scrollViewDecoration = [[UIView alloc] initWithFrame:CGRectMake(0, 444.667, self.view.frame.size.width, self.view.frame.size.height * 1 / 3)];
    [scrollViewDecoration setBackgroundColor:[UIColor whiteColor]];
    [scrollViewDecoration setAlpha:0.3];
    [self.mainScrollView addSubview:scrollViewDecoration];
    
    UIButton *login = [UIButton buttonWithType:UIButtonTypeCustom];
    [login setTitle:@"Sign Up" forState:UIControlStateNormal];
    [login setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [login setTitleColor:[UIColor blueColor] forState:UIControlStateHighlighted];
    login.frame = CGRectMake(0, 0, 200, 100);
    [login setBackgroundColor:[UIColor whiteColor]];
    [login setAlpha:0.2];
    [login addTarget:self action:@selector(moveToSignUpPage:) forControlEvents:UIControlEventTouchUpInside];
    [login setCenter:CGPointMake(self.view.frame.size.width / 2, self.view.frame.size.height - 120)];
    [self.mainScrollView addSubview:login];
    
    self.id = [[UITextField alloc]initWithFrame:CGRectMake(0, 0, 200, 60)];
    [self.id setCenter:CGPointMake(viewSizeWidth / 2, viewSizeHeight / 3)];
    self.id.font = [UIFont systemFontOfSize:15];
    self.id.textColor = [UIColor whiteColor];
    self.id.textAlignment = NSTextAlignmentCenter;
    self.id.borderStyle = UITextBorderStyleNone;
    self.id.placeholder = @"Name";
    self.id.delegate = self;
    self.id.tag = 100;
    [self.mainScrollView addSubview:self.id];
    
    self.pw = [[UITextField alloc]initWithFrame:CGRectMake(0, 0, 200, 60)];
    [self.pw setCenter:CGPointMake(viewSizeWidth / 2, viewSizeHeight / 3 + 60 + 20)];
    self.pw.font = [UIFont systemFontOfSize:15];
    self.pw.textColor = [UIColor whiteColor];
    self.pw.textAlignment = NSTextAlignmentCenter;
    self.pw.borderStyle = UITextBorderStyleNone;
    self.pw.placeholder = @"Passcode";
    self.pw.delegate = self;
    self.pw.tag = 200;
    [self.mainScrollView addSubview:self.pw];


}

- (void)moveToSignUpPage:(UIButton *)sender{
    
    
    ViewControllerSecond *test =  [self.storyboard instantiateViewControllerWithIdentifier:@"ViewControllerSecond"];
    
    [self.navigationController pushViewController:test animated:YES];
    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 20170617/if_For_Switch_SwitfVersion/if_For_Switch_SwitfVersion/ViewController.swift
//
//  ViewController.swift
//  if_For_Switch_SwitfVersion
//
//  Created by ji jun young on 2017. 6. 17..
//  Copyright © 2017년 Fastcampus. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        var roomCapacity:[String:Int] = ["뱅크시" : 5, "모네" : 10 , "프리다" : 15, "마네" : 20];
        
        //roomCapacity += ["고흐" : 55];
        
        //딕셔너리에 값추가
        roomCapacity["고흐"] = 30;
        roomCapacity["폴락"] = 50;
        
        //딕셔너리 특정 키값에 접근
//        roomCapacity["뱅크시"];
        
        //딕셔너리의 키 값 전부 가져오려면 담을 변수를 하나 만들어야 함...허나 튜플 형식으로 가져온 것이므로 완전한 어레이 형태로 가져오려면 다른 방법을 써야...
        let roomNames = roomCapacity .keys;
        let capacity = roomCapacity .values;
        
        print("룸네임스 \(roomNames)")
        print("카파시티 \(capacity)")
        
        //완전한 어레이 형태로 가져오기
        
        let roomNameModified = [String](roomCapacity .keys);
        let capacityModified = [Int](roomCapacity .values);
        
        print("룸 네임 모디파이드 \(roomNameModified)")
        print("카파시티 모디파이드 \(capacityModified)")
        
        // Set 만들기...합집합, 교집합 등 활용해야 할때 사용 가능
        
        let subway2 : Set = ["시청", "을지로3가", "동대문역사문화공원", "홍대입구", "신대방", "강남"];
        let subway3 : Set = ["을지로3가", "동대문역사문화공원", "충무로", "안국", "강남", "경복궁"];
        
        let transfer = subway2.intersection(subway3);
        let nontransfer = subway2.subtracting(subway3);
        let union = subway2 .union(subway3);
        let symm = subway2 .symmetricDifference(subway3);
        
        print("2호선을 기준으로 3호선으로 환승 안되는 역 \(nontransfer)")
        print("2호선과 3호선의 합 \(union)")
        print("환승역이 아닌 곳 \(symm)")
        
        // if, switch, for
        
        if transfer.count > 0 {
            print("환승역은 \(transfer) 입니다")
        }else{
            print("환승역은 없습니다")
        }
        
        for station2 in subway2{
            print("이번 역은 \(station2) 입니다")
            
        }
        
        for station3 in subway3{
            print("이번 역은 \(station3) 입니다")
            
        }
        
        // roomCapacity의 값을 튜플로 받아와서 활용한다
        for (roomName, capacity) in roomCapacity{
            let roomDetail:String
            switch capacity {
            case 5:
                roomDetail = "Room \(roomName) 스터디룸입니다. 정원은 \(capacity) 입니다"
            //스위프트 스위치 구문의 범위 설정 : 숫자...숫자
            case 6...20:
                roomDetail = "Room \(roomName) 은 회의실 입니다. 정원은 \(capacity) 입니다"
            case 30:
                roomDetail = "Room \(roomName)은 강의실 입니다. 정원은 \(capacity) 입니다"
            case _ where capacity > 30 :
                roomDetail = "Room \(roomName)은 대회의실입니다. 정원은 \(capacity) 입니다"
            default:
                roomDetail = "Room 정보를 다시 확인하세요"
            }
            
            print("룸 디테일 \(roomDetail)")
            
        }
        
        // 맥주 수량에 따라 다르게 지시하기
        
        typealias shoppingItem = (name:String, amount:Int)
        
        let cart = shoppingItem("beer", 200)
        
        switch cart {
        case ("beer", 1...3):
            print("소량 계산대로 가세요")
        case ("beer", 4...50):
            print("일반 계산대로 가세요")
        case ("beer", 51...100):
            print("매니저에게 문의하세요")
        case ("beer", _) where cart.amount > 100:
            print("경찰에게 신고하세요")
        default:
            print("줄을 서서 기다리세요")
        }
        
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

